.github/workflows/test.yml

```
@@ -0,0 +1,35 @@
+name: test
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    paths:
+      - '**.go'
+
+jobs:
+  go_test:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.16
+        
+    - name: Install goveralls
+      run: go install github.com/mattn/goveralls@latest
+        
+    - name: Install nmap
+      run: sudo apt install -y nmap
+      
+    - name: Test
```

WDYT of adding some extra sanity checks such as formatting and go mod tidy?

```suggestion
      # This check makes sure that the `go.mod` and `go.sum` files for Go
      # modules are always up-to-date.
      - name: Verify Go modules
        run: go mod tidy && git status && git --no-pager diff && git diff-index --quiet HEAD --

      # This check makes sure that the source code is formatted according to the
      # Go standard `go fmt` formatting.
      - name: Verify source code formatting
        run: go fmt ./... && git status && git --no-pager diff && git diff-index --quiet HEAD --

    - name: Test
```

---

I think it is a good idea to prevent PRs from being merged that do not apply to the Go standards.

Maybe we should split up these checks to two independent jobs:
- This would not delay the test run when the code isn't formatted correctly.